packages/sdk/src/credential/teamsUserCredential.browser.ts

```diff
@@ -178,21 +184,63 @@ export class TeamsUserCredential implements TokenCredential {
       internalLogger.info("Get access token with scopes: " + scopeStr);
-      const cachedKey = await this.getAccessTokenCacheKey(scopeStr);
-      const cachedToken = this.getTokenCache(cachedKey);
-
-      if (cachedToken) {
-        if (!this.isAccessTokenNearExpired(cachedToken)) {
-          internalLogger.verbose("Get access token from cache");
-          return cachedToken;
-        } else {
-          internalLogger.verbose("Cached access token is expired");
+
+      if (!this.initialized) {
+        await this.init();
+      }
+
+      // Get domain from login start page. Will be used in redirect uri.
+      let domain;
+      try {
+        domain = new URL(this.config.initiateLoginEndpoint!).origin;
+      } catch (error: any) {
+        const failedToParseLoginEndpoint = `Failed to parse INITIATE_LOGIN_ENDPOINT in config. Reson: ${error.message}`;
+        internalLogger.error(failedToParseLoginEndpoint);
+        throw new ErrorWithCode(failedToParseLoginEndpoint, ErrorCode.InternalError);
+      }
+
+      let tokenResponse;
+      let errorMessage = "";
+      const scopesArray = typeof scopes === "string" ? scopes.split(" ") : scopes;
+
+      // First try to get Access Token from cache.
+      try {
+        const account = this.msalInstance!.getAccountByUsername(this.loginHint!);
+        const scopesRequestForAcquireTokenSilent = {
+          scopes: scopesArray,
+          account: account ?? undefined,
+          redirectUri: `${domain}/blank-auth-end.html`,
+        };
+        tokenResponse = await this.msalInstance!.acquireTokenSilent(
+          scopesRequestForAcquireTokenSilent
+        );
+      } catch (error: any) {
+        const acquireTokenSilentFailedMessage = `Failed to call acquireTokenSilent. Reason: ${error?.message}. `;
+        errorMessage += acquireTokenSilentFailedMessage;
+      }
+
+      if (!tokenResponse) {
+        // If fail to get Access Token from cache, try to get Access token by silent login.
+        try {
+          const scopesRequestForSsoSilent = {
+            scopes: scopesArray,
+            loginHint: this.loginHint,
+            redirectUri: `${domain}/blank-auth-end.html`,
+          };
+          tokenResponse = await this.msalInstance!.ssoSilent(scopesRequestForSsoSilent);
+        } catch (error: any) {
+          const ssoSilentFailedMessage = `Failed to call ssoSilent. Reason: ${error?.message}. `;
+          errorMessage += ssoSilentFailedMessage;
         }
-      } else {
-        internalLogger.verbose("No cached access token");
       }
 
-      const accessToken = await this.getAndCacheAccessTokenFromSimpleAuthServer(scopeStr);
+      if (!tokenResponse) {
+        const errorMsg = `Get empty authentication result from MSAL. Error: ${errorMessage}`;
+        internalLogger.error(errorMsg);
+        throw new ErrorWithCode(errorMsg, ErrorCode.InternalError);
+      }
+
+      const accessToken = parseAccessTokenFromAuthCodeTokenResponse(JSON.stringify(tokenResponse));
```

Comment:
can we return toeknResponse.accessToken directly here?

Reply:
Update the parseAccessTokenFromAuthCodeTokenResponse, thx！